Build failure: nodejs on/for ppc64

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

On 64-bit big-endian POWER hardware, try building `nodejs` (currently pointing to `nodejs_22`).

On any other system, try building `pkgsCross.ppc64-elfv1.nodejs`.

### Can Hydra reproduce this build failure?

Hydra doesn’t try to build the package.

### Link to Hydra build job

_No response_

### Relevant log output

```console
[2040/2099] LINK mksnapshot
[2041/2099] STAMP obj/tools/v8_gypfiles/v8_snapshot.actions_depends.stamp
[2042/2099] ACTION generating: "obj/tools/v8_gypfiles/v8_snapshot.gen/snapshot.cc" "obj/tools/v8_gypfiles/v8_snapshot.gen/embedded.S"
FAILED: [code=267] obj/tools/v8_gypfiles/v8_snapshot.gen/snapshot.cc obj/tools/v8_gypfiles/v8_snapshot.gen/embedded.S 
cd ../../tools/v8_gypfiles; /nix/store/bdm85np0dj2wdhqrgggypydrkjwh2ybh-qemu-user-10.1.2/bin/qemu-ppc64 /build/node-v22.20.0/out/Release/mksnapshot --turbo_instruction_scheduling --stress-turbo-late-spilling "--target_os=linux" "--target_arch=ppc64" --startup_src /build/node-v22.20.0/out/Release/obj/tools/v8_gypfiles/v8_snapshot.gen/snapshot.cc --embedded_variant Default --embedded_src /build/node-v22.20.0/out/Release/obj/tools/v8_gypfiles/v8_snapshot.gen/embedded.S --no-native-code-counters
qemu: uncaught target signal 11 (Segmentation fault) - core dumped
ninja: build stopped: subcommand failed.
```

### Additional context

Making a build with `--debug` added to `configureFlags`, and running the Debug version of `mksnapshot`, fails some debug assertions related to page sizes. Unsure if related: https://github.com/NixOS/nixpkgs/issues/458983#issuecomment-3504184857

Backtrace of the crash, without the assertions enabled: https://github.com/NixOS/nixpkgs/issues/458983#issuecomment-3504147552

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.17.3, NixOS, 25.11 (Xantusia), 25.11pre880095.5e2a59a5b1a8`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.93.3
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/puna/.config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/puna/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/puna/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/puna/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf:/nix/store/v9g3jiyip4zqfsmalwzglsl2mn1fjxm2-gnome-settings-daemon-48.1/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/zxl9ivn2v7xcrzlf4j036fl6av34zwv9-lix-2.93.3/share`
 - channels(bt1cn): `"unstable"`
 - channels(root): `"nixos, nixos-25.05-25.05, unstable"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

### Notify maintainers

@aduh95

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/web/nodejs/nodejs.nix
+++ b/pkgs/development/web/nodejs/nodejs.nix
@@ -556,7 +556,12 @@ let
         changelog = "https://github.com/nodejs/node/releases/tag/v${version}";
         license = lib.licenses.mit;
         maintainers = with lib.maintainers; [ aduh95 ];
-        platforms = lib.platforms.linux ++ lib.platforms.darwin ++ lib.platforms.freebsd;
+        # https://github.com/nodejs/node/blob/732ab9d658e057af5191d4ecd156d38487509462/BUILDING.md#platform-list
+        platforms =
+          (lib.lists.intersectLists (
+            lib.platforms.linux ++ lib.platforms.darwin ++ lib.platforms.freebsd
+          ) lib.platforms.littleEndian)
+          ++ [ "s390x-linux" ];
         # This broken condition is likely too conservative. Feel free to loosen it if it works.
         broken =
           !canExecute && !canEmulate && (stdenv.buildPlatform.parsed.cpu != stdenv.hostPlatform.parsed.cpu);

